pkg/cni/plugin_test.go

```diff
@@ -0,0 +1,195 @@
+package cni_test
+
+import (
+	"fmt"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/containernetworking/plugins/pkg/testutils"
+	"github.com/kubevirt/macvtap-cni/pkg/cni"
+	"github.com/kubevirt/macvtap-cni/pkg/util"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vishvananda/netlink"
+)
+
+const LOWER_DEVICE = "eth0"
+
+var _ = Describe("Macvtap device plugin", func() {
+	var originalNS ns.NetNS
+	var targetNs ns.NetNS
+	var lowerDevice netlink.Link
+	var macvtapInterface netlink.Link
+	var deviceID string
+	var stdInArgs string
+
+	BeforeEach(func() {
+		var err error
+		originalNS, err = testutils.NewNS()
+		Expect(err).NotTo(HaveOccurred())
+		targetNs, err = testutils.NewNS()
+		Expect(err).NotTo(HaveOccurred())
+
+		err = originalNS.Do(func(ns.NetNS) error {
+			defer GinkgoRecover()
+
+			// create lower device
+			err = netlink.LinkAdd(&netlink.Dummy{
+				LinkAttrs: netlink.LinkAttrs{
+					Name: LOWER_DEVICE,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			lowerDevice, err = netlink.LinkByName(LOWER_DEVICE)
```

Comment:
Let it breathe:

```
action
error handling
<space>
```

Please apply it in the whole test

Reply:
Done.